### ISSUE TITLE
Using “prettier-ignore” comments does not work as expected

### ISSUE THREAD

**Issue by tylervz** [eemeli/prettier-plugin-properties#12]
As the [Prettier documentation states](https://prettier.io/docs/en/ignore),  users can use “prettier-ignore” comments to ignore parts of files. This works for the file types officially supported by Prettier such as JavaScript, HTML and CSS.

I was hoping it would work for my properties files as well, but it did not work as I had hoped. I'm using prettier version 3.0.3 and prettier-plugin-properties version 0.2.0. My `.prettierrc.json` file is as follows:

```json
{
    "singleQuote": false,
    "overrides": [
        {
            "files": ["*.properties"],
            "options": {
                "keySeparator": "=",
                "printWidth": 100
            }
        }
    ],
    "plugins": ["prettier-plugin-properties"]
}
```

I have an `example.properties` file that looks like so:

```text
instructorCoordinationService.recordESigAttendance.learner.notFound=The user identification \
    entered, {0}, did not match any learners in the system.

# prettier-ignore
instructorCoordinationService.recordESigAttendanceCollectedOffline.sessionItem.notFound.error=SessionItem with id {0} not found

instructorCoordinationService.recordESigAttendanceCollectedOffline.attendee.added={0} {1} added
```
Running `npx prettier --write example.properties` results in a newline being removed:

```text
instructorCoordinationService.recordESigAttendance.learner.notFound=The user identification \
    entered, {0}, did not match any learners in the system.

# prettier-ignore
instructorCoordinationService.recordESigAttendanceCollectedOffline.sessionItem.notFound.error=SessionItem with id {0} not found
instructorCoordinationService.recordESigAttendanceCollectedOffline.attendee.added={0} {1} added
```

Running `npx prettier --write example.properties` again results in the third key/value pair being appended to the second key/value pair:

```text
instructorCoordinationService.recordESigAttendance.learner.notFound=The user identification \
    entered, {0}, did not match any learners in the system.

# prettier-ignore
instructorCoordinationService.recordESigAttendanceCollectedOffline.sessionItem.notFound.error=SessionItem with id {0} not foundinstructorCoordinationService.recordESigAttendanceCollectedOffline.attendee.added={0} {1} added
```

I expected the file to not be modified after running `npx prettier --write example.properties` subsequent times.

Looking briefly at the source code for this plugin, it looks like it is checking for prettier-ignore comments.

https://github.com/eemeli/prettier-plugin-properties/blob/0c7072dae2006c0d0a588f70b19940390a50add1/index.js#L39-L46

I can tell it does behave differently with the prettier-ignore comment being present in my file, just not the way I hoped.

Update: in my fork of the plugin, I've added [another test case](https://github.com/tylervz/prettier-plugin-properties/commit/dda4c9ec88da35ba3f5711cd6e5b721963ed5ebd) for scenarios such as mine, however I haven't yet been able to modify `index.js` so that the test passes.

**Comment by eemeli** [comment]
Huh, it does look like the trailing newline is getting lost for the key-value pair after a `# prettier-ignore`.

A PR fixing this would be very welcome; I'm not sure when I might get to this myself.

**Comment by tylervz** [comment]
I have looked into fixing this and, from what I can tell, the issue does not seem to reside in the `hasPrettierIgnore` function.

https://github.com/eemeli/prettier-plugin-properties/blob/0c7072dae2006c0d0a588f70b19940390a50add1/index.js#L34-L47

Looking at how Prettier handles the `hasPrettierIgnore` function of a printer, I think the issue might  reside in the `locEnd` function of `prettier-plugin-properties`. In the cases where `printIgnored` is being called, I think if `locEnd` returned a value 1 greater than what it currently does, then the trailing newline would not be removed.

https://github.com/prettier/prettier/blob/d0b907620b5a59fb75829e1b46e0636a064adae6/src/main/ast-to-doc.js#L101-L102

https://github.com/prettier/prettier/blob/d0b907620b5a59fb75829e1b46e0636a064adae6/src/main/print-ignored.js#L11-L20

https://github.com/eemeli/prettier-plugin-properties/blob/0c7072dae2006c0d0a588f70b19940390a50add1/index.js#L15-L19

I still need to do some fiddling around with the code to try getting it to work.

**Comment by tylervz** [comment]
@eemeli I was able to figure it out and open a pull request!

**Comment by eemeli** [comment]
Fixed by #13, and included in release 0.3.0.
